packages/story-editor/src/app/media/utils/useDetectBlurhash.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from '@web-stories-wp/react';
+import { getBlurFromImage } from '@web-stories-wp/media';
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../api';
+import { useStory } from '../../story';
+import { useConfig } from '../../config';
+import useCORSProxy from '../../../utils/useCORSProxy';
+
+function useDetectBlurHash({ updateMediaElement }) {
+  const {
+    actions: { updateMedia },
+  } = useAPI();
+  const { updateElementsByResourceId } = useStory((state) => ({
+    updateElementsByResourceId: state.actions.updateElementsByResourceId,
+  }));
+  const {
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+  const { getProxiedUrl } = useCORSProxy();
+  const setProperties = useCallback(
+    (id, properties) => {
+      updateElementsByResourceId({ id, properties });
+    },
+    [updateElementsByResourceId]
+  );
+
+  const saveBlurHash = useCallback(
+    /**
+     *
+     * @param {number} id Video ID.
+     * @param {string} blurHash Base Color.
+     * @return {Promise<void>}
+     */
+    async (id, blurHash) => {
+      try {
+        const newState = ({ resource }) => ({
+          resource: {
+            ...resource,
+            blurHash,
+          },
+        });
+        setProperties(id, newState);
+        updateMediaElement({
+          id,
+          data: { blurHash },
+        });
+        if (hasUploadMediaAction) {
+          await updateMedia(id, {
+            meta: { web_stories_blurhash: blurHash },
+          });
+        }
+      } catch (error) {
+        // Do nothing for now.
+      }
+    },
+    [setProperties, updateMedia, updateMediaElement, hasUploadMediaAction]
+  );
+
+  const updateBlurHash = useCallback(
+    async ({ resource }) => {
+      const { type, src, poster } = resource;
+      const imageSrc = type === 'image' ? src : poster;
+      if (!imageSrc) {
+        return;
+      }
+      const imageSrcProxied = getProxiedUrl(resource, imageSrc);
```

Comment:
Media items in the media library never need proxying

Hotlinked images never need blurhashes

Let's remove this.

Reply:
This isn't true. My personal site has access sorted in S3 and they do need proxying to work.

Reply:
You should set up CORS headers on your S3 bucket then.

I know it's convenient, but the CORS proxy was not meant to be used like this.

It is designed for use with hotlinking only.

Using it for _everything_ is going to have an unnecessary and huge performance impact on your server.

Reply:
Do we nearly expect everyone to do this? The average user will not know how to do this...

Reply:
Yes.

I know that in support this happens very very rarely and can be resolved very quickly every time.
Part of the reason is that CDNs are usually set up by more technical people, so they know how to fix it.

I understand your concern, I really do.

If circumstances should change in the future (e.g. the number of support requests increasing significantly), this can be revisited.